src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PopupWindow.cs

```diff
@@ -0,0 +1,124 @@
+using System;
+using UnityEditor;
+using UnityEngine;
+
+namespace GitHub.Unity
+{
+    [Serializable]
+    class PopupWindow : BaseWindow
+    {
+        public enum PopupViewType
+        {
+            PublishView,
+            AuthenticationView
+        }
+
+        [NonSerialized] private Subview activeView;
+
+        [SerializeField] private PopupViewType activeViewType;
+        [SerializeField] private AuthenticationView authenticationView;
+        [SerializeField] private PublishView publishView;
+
+        [MenuItem("GitHub/Authenticate")]
+        public static void Launch()
+        {
+            var popupWindow = Open(PopupViewType.AuthenticationView);
+            popupWindow.InitializeWindow(EntryPoint.ApplicationManager);
+        }
+
+        public static PopupWindow Open(PopupViewType popupViewType, Action<bool> onClose = null)
+        {
+            var popupWindow = GetWindow<PopupWindow>(true);
+            if (onClose != null)
+            {
```

Comment:
The `Open` and `InitializeWindow` stuff was no problem...

I made `RaiseOnClose` virtual and calling a method to clear the `OnClose` handler In the `PopupWindow` override.

https://github.com/github-for-unity/Unity/blob/d4121b20ae7e218650a31016b4f7944cc8fa095d/src/UnityExtension/Assets/Editor/GitHub.Unity/UI/PopupWindow.cs#L86-L90